PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-java-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/java/sdk/models/DependencyVersionsDataCache.java

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.azure.toolkit.intellij.java.sdk.models;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serial;
+import java.io.Serializable;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+/**
+ * A cache to store the dependency versions data for a certain period of time.
+ * The cache is stored in memory and also saved to a file to persist the data across IDE restarts.
+ * The cache is cleared at regular intervals to prevent it from growing indefinitely.
+ *
+ * @param <T> The type of the data to be stored in the cache.
+ */
+public class DependencyVersionsDataCache<T> implements Serializable {
```

Comment:
We should reuse the `CacheManager` provided by the common lib instead of having our own caching implementation.

Reply:
https://github.com/microsoft/azure-tools-for-java/pull/9654#discussion_r1935080966